### PR TITLE
Lesson

### PR DESCRIPTION
**Pull request by kadikraman**
Work in progress - I've made a PR as it's easier to keep track

- [x] - componentWillMount (& test)
- [x] - componentDidMount (& test)
- [x] - componentWillUnmount (& test)
- [x] - score
- [x] - instruction text in readme
- [x] - some basic styling

### CODE REVIEW

**Comment by kadikraman** [comment]
@thomastuts I'm almost done with this, but I can't think of anything to do in `componentWillMount` - do you have any ideas?

File: README.md [hunk 1]
```diff
@@ -10,9 +10,38 @@
...
+
+Go on and run the game. It doesn't quite work yet, because some essential bits are missing and it's up to you to finish it off!
+
+Pancakes need to be cooked equally on both sides to be yummy. The current settings are that if a pancake is cooked for
+exactly 2 seconds on either side, it's considered to be cooked perfectly. If it's cooked for more than 2 seconds on either
+side it will be burnt and otherwise it will be raw. You can change these settings in `Pancake.js`!
+
+## Shop needs an opening time
+First, you want to record the time your shop was opened. Open `Game.js` and notice a `setCurrentTime()` method already exits,
+but it is not yet called. Create a `componentWillMount()` in `Game.js` which will call the `setCurrentTime()` when the
```
Comment (thomastuts) [inline README.md]: > Create a `componentWillMount()` **method** in...

File: README.md [hunk 2]
```diff
@@ -10,9 +10,38 @@
...
+component is first created. If you've done that correctly, you should see the current time at the top of the page.
+
+## Pancake needs a timer!
+
+Now it's time to make your pancakes work. Firstly, we need a timer on each pancake so we could record how long it's
```
Comment (thomastuts) [inline README.md]: 'Make pancakes work' sounds a little off, can we rephrase this?

File: README.md [hunk 3]
```diff
@@ -10,9 +10,38 @@
...
+been cooking. Look in `Pancake.js` - the `startInterval()` bethod is already there. All you need to do it add a lifecycle
```
Comment (thomastuts) [inline README.md]: 'bethod' -> 'method'

'it' -> 'is'

File: README.md [hunk 4]
```diff
@@ -10,9 +10,38 @@
...
+method in `Pancake.js` at `componentDidMount()` (that's the point at which the pancake component gets added to the page)
```
Comment (thomastuts) [inline README.md]: Using capitals and backticks for names when talking about components, e.g. `Pancake` or `<Pancake />` is more consistent with the rest of the course content

File: README.md [hunk 5]
```diff
@@ -10,9 +10,38 @@
...
+which will start the counter.
+
+
+## Tidy up the timer
+
+Now that we've set up a timer, we need to make sure we also remove it as soon as it's not needed anymore. It's important
+to always do these these two in pairs. As you can see, a `cleanUpInterval()` function is all set up for you so all you
```
Comment (thomastuts) [inline README.md]: 'to always do these these two in pairs' sounds a little strange, I assume you mean that you should always destroy a timer/interval when creating one? Can we rephrase this a little?

File: README.md [hunk 6]
```diff
@@ -10,9 +10,38 @@
...
+need to do it call it just before the component gets unmouned from the page. That will be in the `componentWillUnmount()`
```
Comment (thomastuts) [inline README.md]: 'unmouned' -> 'unmounted'

**Comment by thomastuts** [comment]
Thanks @kadikraman, this looks great!

**Pull request merged**